platforms/pegasus/motion_control.cpp

```diff
@@ -0,0 +1,339 @@
+    linear_dualpid_meta_controller.add_controller(&linear_speed_controller);
+    std::cout << "SpeedPIDController created and added to LinearDualPIDMetaController" << std::endl;
+
+    // Angular dual PID meta controller
+    static cogip::motion_control::DualPIDMetaController angular_dualpid_meta_controller;
+    polar_parallel_meta_controller.add_controller(&angular_dualpid_meta_controller);
+    std::cout << "AngularDualPIDMetaController created and added to PolarParallelMetaController" << std::endl;
+    static cogip::motion_control::PosePIDControllerParameters angular_position_controller_parameters(&angular_pose_pid);
+    static cogip::motion_control::PosePIDController angular_position_controller(&angular_position_controller_parameters);
+    angular_dualpid_meta_controller.add_controller(&angular_position_controller);
+    std::cout << "PosePIDController created and added to AngularDualPIDMetaController" << std::endl;
+    // Angular speed filter
+    static cogip::motion_control::SpeedFilterParameters angular_speed_filter_parameters(
+        cogip::pf::motion_control::platform_max_speed_angular_deg_per_period,
+        cogip::pf::motion_control::platform_max_acc_angular_deg_per_period2
+        );
+    static cogip::motion_control::SpeedFilter angular_speed_filter(&angular_speed_filter_parameters);
+    angular_dualpid_meta_controller.add_controller(&angular_speed_filter);
+    std::cout << "SpeedFilter created and added to AngularDualPIDMetaController" << std::endl;
+    static cogip::motion_control::SpeedPIDControllerParameters angular_speed_controller_parameters(&angular_speed_pid);
+    static cogip::motion_control::SpeedPIDController angular_speed_controller(&angular_speed_controller_parameters);
+    angular_dualpid_meta_controller.add_controller(&angular_speed_controller);
+    std::cout << "SpeedPIDController created and added to AngularDualPIDMetaController" << std::endl;
+
+    // Quad PID meta controller
+    static cogip::motion_control::QuadPIDMetaController quadpid_meta_controller;
+    quadpid_meta_controller.add_controller(&pose_straight_filter);
+    quadpid_meta_controller.add_controller(&polar_parallel_meta_controller);
+
+    return &quadpid_meta_controller;
+}
+
+void pf_print_state(void)
+{
+    COGIP_DEBUG_COUT(
+        "{"
+            << "\"pose_current\":{"
+                << "\"O\":" << pf_motion_control_platform_engine.current_pose().O()
+                << ",\"x\":" << pf_motion_control_platform_engine.current_pose().x()
+                << ",\"y\":" << pf_motion_control_platform_engine.current_pose().y()
+            << "},"
+            << "\"cycle\":" << pf_motion_control_platform_engine.current_cycle() << ","
+            << "\"speed_current\":{\"distance\":" << pf_motion_control_platform_engine.current_speed().distance()
+            << ",\"angle\":" << pf_motion_control_platform_engine.current_speed().angle() << "}"
+    );
+
+    COGIP_DEBUG_COUT("}");
```

Comment:
This must be included in the previous statement, but I think this function can be removed.

Reply:
done

Reply:
can and should be removed but in a next PR ;)